### ISSUE TITLE
Document that pipelines implicitly do clearing

### ISSUE THREAD

**Issue by kpreid** [phaazon/luminance-rs#467]
I wanted to add clearing the depth buffer between two rendering steps. I first found `Buffer::clear`, and concluded that this wasn't helpful because it would require borrowing the `Buffer` that the pipeline already had borrowed. I then noticed that a docs search mostly turned up `PipelineState` talking about clearing, and realized that it must be the case that each `Pipeline` execution performs a clear operation as its first effect, and so in order to do what I wanted I needed to call `surface.new_pipeline_gate()` twice.

This suggests two things. First, I don't know how expensive a Pipeline by itself is, but if it is significant, then perhaps it should be possible to clear without having to exit the pipeline.

Second, the documentation for [PipelineGate::pipeline](https://docs.rs/luminance/0.43.1/luminance/pipeline/struct.PipelineGate.html#method.pipeline) and [Pipeline](https://docs.rs/luminance/0.43.1/luminance/pipeline/struct.Pipeline.html) should specify explicitly that the _effects_ of employing a pipeline are to clear the buffer as controlled by the `PipelineState`, in addition to any drawing specified explicitly in the closure.

**Comment by phaazon** [comment]
Have you checked [this](https://docs.rs/luminance/0.43.1/luminance/pipeline/struct.PipelineState.html#structfield.clear_depth_enabled)?

**Comment by phaazon** [comment]
I agree pipelines could get more documentation. An empty pipeline will do pretty much nothing interesting, besides what the default state suggest. It will create a bunch of objects on the stack (just a bunch of references for most part) and call the start pipeline method of the backend, which will inspect the `PipelineState` and will perform the operations according to what is set there.

**Comment by kpreid** [comment]
> Have you checked this?

Yes, I found the PipelineState information, but that's only informative _after one has found the place to look._ When I started from "I want to _add_ some clearing", it didn't readily occur to me that the way to do that would be to create a second Pipeline; if this was documented in more places it would be easier to find.

> An empty pipeline will do pretty much nothing interesting, besides what the default state suggest.

Yes, but "pipeline _state_" suggests _options_ that might influence future operations, not that it directly specifies the operation of clearing.

To rephrase the idea, aiming at the bigger picture: one part of the luminance docs (I couldn't quickly find it again) described luminance in terms of "AST nodes". I'm thinking that each type/function that is about a node type could briefly summarize the role _that node type_ plays in drawing, and what its "parent" and "child" nodes are; in the case of pipelines, the summary would be something like "clear the framebuffer as specified in the PipelineState, and allow binding textures and buffers in preparation for a `ShadingGate`".
